### DISCUSSION TITLE
Lesson 7: A confusion about the usage of parameters of a contract

### DISCUSSION THREAD

**Discussion by TreapGoGo** [Cyfrin/foundry-full-course-cu#319] [Q&A] [answered]
In `Interactions.s.sol` we have 

```solidity
FundMe(payable(mostRecentlyDeployed)).fund{value: SEND_VALUE}();
```

in the function `fundFundMe`, where the parameter `payable(mostRecentlyDeployed)` acts as an indication of an address, where an `FundMe` contract has been deployed, according to [@PatrickAlphaC's answer](https://github.com/Cyfrin/foundry-full-course-f23/discussions/296#discussioncomment-6393675). It seems that `FundMe()` here can be regarded as a function and it just converts the type of the parameter (i.e. from `address` to `FundMe` contract). It doesn't create a new contract but locate an existing contract at a specific address.

However, in `DeployFundMe.s.sol` we have

```solidity
FundMe fundMe = new FundMe(ethUsdPriceFeed);
```

in the function `run`, where the parameter `ethUsdPriceFeed` is sent to the `constructor` of `FundMe` in order to generate an interface of price feeding, just like other Objective-Oriented languages.

These usages makes me confused :(

I would appreciate it if someone could explain this to me. :)

**Comment by shikhar229169** [comment, marked as answer]
So when we write new keyword followed by contract name, it means that we are deploying a new contract, and we need to pass the constructor arguments.

But when we simple write it without using new keyword it means that we are simply type casting the address to the contract.
```Solidity
FundMe fundme = FundMe(address)
```
This means that we are converting the address type to contract type and the instance is stored in fundme.

**Reply by TreapGoGo** [reply]
Thanks a lot for your explanation! Have a good day!
